### PR TITLE
PERF-5111 Multiplanner simple case

### PR DESCRIPTION
**Pull request by dhsegel**

### CODE REVIEW

File: src/workloads/query/multiplanner/Simple.yml
Comment (dpercy) [inline src/workloads/query/multiplanner/Simple.yml:239]: We should probably split this actor into three: one for each phase, so that when we use a different number of indexes the metrics go to a different OperationMetricsName.

On this patch it looks like they're being aggregated when I filter by RunQueries, AverageLatency there's only on row: https://spruce.mongodb.com/version/65de69655623437ddd50da76/tasks?page=0&sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC%3BNAME%3AASC

The numbers I got so far are:
| build variant | latency (not sure what unit) |
|--------|--------|
| standalone | 21.9 million |
| standalone all feature flags | 22.5 million |
| standalone classic | 22.1 million | 

Doesn't it seem weird that these numbers would be so close?  The SBE multiplanner should be worse, because we have 1 good plan and 62 bad plans.  I wonder if it's getting lucky, and enumerating the good plan first.  If it limits each trial execution by the best upper bound so far, then seeing the best plan first lets it avoid wasting time on the bad plans.

File: src/workloads/query/multiplanner/Simple.yml
Comment (Adityav369) [inline src/workloads/query/multiplanner/Simple.yml:69]: Let's move this part to `src/phases/..`. This way, for future tickets like PERF-5112, PERF-5113 we can import the InsertActor and just make the different indexes. 

You could make the indexes field a `^Parameter` and the whole actor can be moved. 
See `src/phases/query/AggregateExpressions.yml` for example.

File: src/workloads/query/multiplanner/Simple.yml
Comment (Adityav369) [inline src/workloads/query/multiplanner/Simple.yml:230]: A cleaner way would be to use OnlyActiveInPhases like the other phases and then redefine the Actor for phase 5 and 7 with different names like `SimpleMultiplannerWithXIndexes`. This way you can also define a find query template and parameterize the `Active` and `Name`.

**Review by Adityav369 — CHANGES_REQUESTED** [review]
Thanks Daniel, I added a couple comments. Lmk if you have questions.

File: src/workloads/query/multiplanner/Simple.yml
Comment (dpercy) [inline src/workloads/query/multiplanner/Simple.yml:47]: I think this actor will be identical in the other scenarios, so it can be pulled out into a separate file.

File: src/workloads/query/multiplanner/Simple.yml
Comment (dpercy) [inline src/workloads/query/multiplanner/Simple.yml:321]: It would be helpful to make these names more meaningful, like "query with 63 indexes".

Even though you still need to understand the workload to understand the output, having clearer labels would reduce amount of flipping back and forth between the output and workload definition.

File: src/workloads/query/multiplanner/Simple.yml
Comment (dpercy) [reply src/workloads/query/multiplanner/Simple.yml]: Ah, this must be because all three are using the classic multiplanner:
* standalone, the default configuration = classic+classic
* standalone all feature flags = classic+SBE, now that our project's feature flag is included here
* standalone classic = classic+classic

So I'll add this test to another build variant, for standalone-sbe.

File: src/workloads/query/multiplanner/Simple.yml
Comment (Fefer-Ivan) [inline src/workloads/query/multiplanner/Simple.yml:416]: This is not SBE
This is SBE with no other feature flags
```
standalone-sbe
```

File: src/workloads/query/multiplanner/Simple.yml
Comment (Fefer-Ivan) [inline src/workloads/query/multiplanner/Simple.yml]: You also don't need to specify OperationMetricsName. It is an optional field, so if it is not something meaningful, let's just delete it and it will default to OperationName

File: src/workloads/query/multiplanner/Simple.yml
Comment (dhsegel) [reply src/workloads/query/multiplanner/Simple.yml:322]: Removing altogether per Ivan's comment below. Will keep clear labels in the `Name` field.

File: src/phases/query/Multiplanner.yml
Comment (Adityav369) [inline src/phases/query/Multiplanner.yml:12]: For `default`, we could use incorrect values like a string. This way, we can ensure we do not miss a parameter when using this template. Something like https://github.com/mongodb/genny/blob/master/src/workloads/query/LimitSkip.yml#L34.
This comment applies to other places Parameter is used.

File: src/workloads/query/multiplanner/Simple.yml
Comment (Adityav369) [inline src/workloads/query/multiplanner/Simple.yml:64]: As per another comment in template file, no need for `threadsA` and `threadsB`.

File: src/workloads/query/multiplanner/Simple.yml
Comment (Adityav369) [inline src/workloads/query/multiplanner/Simple.yml:277]: [nit] format

**Review by Adityav369 — CHANGES_REQUESTED** [review]
Thanks Daniel, just added some comments about style. In terms of the logic and actual workload, this looks about right!

File: src/workloads/query/multiplanner/Simple.yml
Comment (dhsegel) [reply src/workloads/query/multiplanner/Simple.yml:277]: What is the problem here? Should `v7.3` be on its own line?

File: src/workloads/query/multiplanner/Simple.yml
Comment (dhsegel) [reply src/workloads/query/multiplanner/Simple.yml:170]: I'm not sure this is possible. I don't believe I can use `^FormatString` in conjunction with `^Inc` to succinctly name all the indexes. Is this similar to what you had in mind?
`- {OperationName: RunCommand, OperationCommand: {collMod: *coll, index: {hidden: true, keyPattern: {{^FormatString: {"{x%d_1", "withArgs:": {^Inc: {start: 17}}}}}}}}`

Comment (dpercy) [reply src/workloads/query/multiplanner/Simple.yml:170]: For example, imagine trying to factor it out like this:

```
place_for_macro_definitions:
  my_macro:
    OperationName: RunCommand
    OperationCommand:
      collMod: *coll
      index: 
        hidden: true
        keyPattern: {^Parameter: {Name: "keyPattern", Default: "invalid"}}
...
Operations:
- {invoke_a_macro: 'my_macro', args: {keyPattern: {x1: 1}}}
- {invoke_a_macro: 'my_macro', args: {keyPattern: {x2: 1}}}
...
- {invoke_a_macro: 'my_macro', args: {keyPattern: {x63: 1}}}
```

There would still be repetition, which we need to verify by eye, but it also adds an indirection that can make it less clear what an individual command looks like.

Comment (dhsegel) [reply src/workloads/query/multiplanner/Simple.yml:170]: That makes sense to me, but is that preferred stylistically? It would be more lines of code and less readable than just repeating the same 64 lines where it is very clear that the same operation is being done to each index.

Comment (dpercy) [reply src/workloads/query/multiplanner/Simple.yml:170]: Right, to be clear I was arguing that it wouldn't be an improvement overall to factor it out that way.

File: src/workloads/query/multiplanner/Simple.yml
Comment (Adityav369) [reply src/workloads/query/multiplanner/Simple.yml:277]: I think there is an extra space on L27. There is a red circle which indicates that a blank line is not left at the end of file (which is standard practice for git). If you remove this space, it should go away.

File: src/workloads/query/multiplanner/Simple.yml
Comment (Adityav369) [reply src/workloads/query/multiplanner/Simple.yml:170]: That's an interesting approach, and I agree with @dpercy that it is better to be explicit. I was thinking of something similar to [this](https://github.com/mongodb/genny/pull/1134#discussion_r1513153261), but it is not important. Feel free to ignore this suggestion, better to be explicit.

File: src/workloads/query/multiplanner/Simple.yml
Comment (Adityav369) [inline src/workloads/query/multiplanner/Simple.yml:155]: This is no longer relevant, right? Since you have extracted the query from the template and are supplying the selectivity value directly? I also think that the predicate on x1 will default to 1 in this case. We can now directly use selectivity.

**Review by Adityav369 — APPROVED** [review]
Thanks Daniel! LGTM from my side, but I left one last comment. This looks great, will make the subsequent tickets much easier to implement. 

I have no further comments as of now and will defer to the assignee for the final LGTM.

File: src/workloads/query/multiplanner/Simple.yml
Comment (dhsegel) [reply src/workloads/query/multiplanner/Simple.yml:155]: Thanks for catching that!

**Review by dpercy — APPROVED** [review]

File: src/workloads/query/multiplanner/Simple.yml
Comment (Adityav369) [reply src/workloads/query/multiplanner/Simple.yml]: @dhsegel pls fix this before you're ready to merge.

Comment (dhsegel) [reply src/workloads/query/multiplanner/Simple.yml]: Done. I had done this to `Simple.yml` like you saw but not to `Multiplanner`.

**Review by Fefer-Ivan — APPROVED** [review]

**Comment by dhsegel** [comment]
evergreen merge

**Pull request merged**
